src/components/popup/popup-manager.spec.tsx

```diff
@@ -0,0 +1,50 @@
+import { mount } from 'enzyme';
+import {Observable, TestScheduler} from 'rxjs';
+import { Popup, PopupProps } from './popup';
+import {PopupManager} from './popup-manager';
+
+describe('PopupManager', () => {
+    let scheduler: TestScheduler;
+
+    beforeEach(() => {
+        scheduler = new TestScheduler((a, b) => expect(a).toEqual(b));
+    });
+
+    describe('confirm', () => {
+        it.each([
+            ['OK', true, '[qe-id="argo-popup-ok-button"]'],
+            ['Cancel', false, '[qe-id="argo-popup-cancel-button"]'],
+        ])('%s', async (_, promiseResult, btnSelector) => {
+            const fn = jest.fn<void, [null | PopupProps]>();
+            const manager = new PopupManager();
+
+            Observable.from(manager.popupProps, scheduler).subscribe(fn);
+            scheduler.flush();
+
+            expect(fn).toHaveBeenCalledTimes(1);
+            expect(fn).toHaveBeenLastCalledWith(null);
+
+            const promise = manager.confirm('foo', 'bar');
+            scheduler.flush();
```

Comment:
I originally had this written with RxJS "marble" tests which are pretty neat, however syntax used for them is pretty strange if you've never used them or seen them, so I chose to long-hand this.  Would be interested to hear others' thoughts:

https://rxjs-dev.firebaseapp.com/guide/testing/marble-testing